handlers/handlers.go

```diff
@@ -28,86 +27,170 @@ type RenderClient interface {
 
 // SearchClient is an interface with methods required for a search client
 type SearchClient interface {
-	GetSearch(ctx context.Context, query url.Values) (r search.Response, err error)
+	GetSearch(ctx context.Context, query url.Values) (r searchC.Response, err error)
 }
 
-func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
-	status := http.StatusInternalServerError
-	if err, ok := err.(ClientError); ok {
-		if err.Code() == http.StatusNotFound {
-			status = err.Code()
-		}
+// Read Handler
+func Read(cfg *config.Config, rendC RenderClient, searchC SearchClient) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		read(w, req, cfg, rendC, searchC)
 	}
-	if err.Error() == "invalid filter type given" {
-		status = http.StatusBadRequest
+}
+
+func read(w http.ResponseWriter, req *http.Request, cfg *config.Config, rendC RenderClient, searchC SearchClient) {
+	ctx := req.Context()
+
+	urlQuery := req.URL.Query()
+
+	validatedQueryParams, err := data.ReviewQuery(ctx, cfg, urlQuery)
+	if err != nil {
+		log.Event(ctx, "unable to review query", log.Error(err), log.ERROR)
+		setStatusCode(w, req, err)
+		return
+	}
+
+	apiQuery := data.GetSearchAPIQuery(validatedQueryParams)
+
+	resp, err := searchC.GetSearch(ctx, apiQuery)
+	if err != nil {
+		logData := log.Data{"api query passed to search-api": apiQuery}
+		log.Event(ctx, "getting search response from client failed", log.Error(err), log.ERROR, logData)
+		setStatusCode(w, req, err)
+		return
+	}
+
+	// TO-DO: Until API handles aggregration on datatypes (e.g. bulletins, article), we need to make a second request
```

Comment:
The second request is carried out in the function `getCategoriesTypesCount` (in `handlers.go` line `104`). 
We cannot remove the comment because right now, the number of search results for `every` filter type is retrieved by doing a second request to the API (in the second request, we are not passing any filters so the search-API provides all the possible search results with count for every type).  

The first request only provides the number of search results for `each` filter type applied by the user

In the future, we want the API to handle aggregation on filter datatypes where we can avoid the second request from controller to api